Transaction/DLC_Funding_Transaction.md

```diff
@@ -0,0 +1,110 @@
+# DLC Funding Transaction
+
+This details the exact format of the on-chain funding transaction, which both sides need to agree on to ensure signatures are valid.
+
+# Table of Contents
+
+* [Transaction](#transaction)
+  * [Transaction Input and Output Ordering](#transaction-input-and-output-ordering)
+  * [Funding Inputs](#funding-inputs)
+  * [FundingOutput](#funding-output)
+  * [Change Outputs](#change-outputs)
+* [Fees](#fees)
+
+# Transaction
+
+## Transaction Input and Output Ordering
+
+The Offerer/Initiator's inputs come first and in the order that they were listed in the Offer message, the Accepter's inputs come after this in the order they are listed in the Accept message. The funding output comes first, followed by the Offerer's change output if applicable, and then the Accepter's change output if applicable.
+
+Note that this will likely change in the future.
+
+## Funding Inputs
+
+All funding inputs must be Segwit or nested P2SH(Segwit) in order to protect against malleability attacks.
```

Comment:
We may have a concern with malleable _witnesses_.

Let's say both Alice and Bob both announce P2WSH inputs. They exchange their inputs, sign the funding transactions, sign CETs timelocked to 1 week. Bob then broadcast the funding transaction with a valid witness deflating the feerate so low that the funding transaction is stuck in network mempools. After a week, Alice wins the contract but can't spend her CET as the funding has never been confirmed. After 2 weeks, when funding transaction expires from network mempools, Bob double-spend his collateral back to him, that way invalidating a loosing contract.

This assume some congestion in mempools. Miniscript might help there. Note, channel doesn't have this issue as absolute timelock aren't starting before the funding transaction.

Reply:
>Bob then broadcast the funding transaction with a valid witness deflating the feerate so low that the funding transaction is stuck in network mempools.
how does this work?

Reply:
> After 2 weeks, when funding transaction expires from network mempools, Bob double-spend his collateral back to him, that way invalidating a loosing contract.

It seems like congested mempools are a general edge case for funding DLC's. Perhaps clients should be designed to double spend in the case that the funding tx is not confirmed within x amount of time by default.

Reply:
> how does this work? Are you saying that some scripts have malleable witnesses so Bob can chuck random crap in their to increase the size of it without Alice knowing after they've agreed to a different witness.

Yes, if my P2WSH witnessScript has control flow embedded (IF/NOTIF/...), I can announce you witness 1 with feerate 10 sat/vByte but broadcast a bigger valid witness 2 dropping feerate of the whole transaction to 1  sat/vByte thus downgrading its confirmation chance in case of high-congestion.

> It seems like congested mempools are a general edge case for funding DLC's. Perhaps clients should be designed to double spend in the case that the funding tx is not confirmed within x amount of time by default.

Either this solution or anchor output. I think we would like anchor outputs anyway to let DLC participants exercise unilateral fee-bumping in case of congestion, independently of malicious behavior. Given, we don't have previous state like LN for onchain DLC, we don't have the issue of blind-CPFP/pinning that package relay solves.

Also, anchor output let a single-party push the DLC state forward, whereas RBF'ing to double-spend after x amount of time, you loss timevalue-wrt-DLC-bid of your collateral during x.

Reply:
Since we are going to have to mitigate witness malleability and fee rate malleability generally, can we allow non-witness inputs or is there some way in which segwit v0 still ends up helping us here?